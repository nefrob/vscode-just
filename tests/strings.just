# Strings

# Escaping sequences

string-with-tab             := "\t"
string-with-newline         := "\n"
string-with-carriage-return := "\r"
string-with-double-quote    := "\""
string-with-slash           := "\\"
string-with-no-newline      := "\
"

escapes := """\t\n\r\"\\"""
escapes := '\t\n\r\"\\'
escapes := '''\t\n\r\"\\'''

# Multi-line

single := '
hello
'

double := "
goodbye
"

# Indented

x := '''
  foo
  bar
'''

y := """
  abc
    wuv
  xyz
"""

# Expanded

foobar := x"~/$FOO/${BAR}"
foobar := x"""~/$FOO/${BAR}"""
foobar := x'~/$FOO/${BAR}'
foobar := x'''~/$FOO/${BAR}'''
