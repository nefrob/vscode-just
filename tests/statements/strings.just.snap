># Strings
#^^^^^^^^^ source.just comment.line.number-sign.just
>
># Escaping sequences
#^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>string-with-tab             := "\t"
#^^^^^^^^^^^^^^^ source.just variable.other.just
#               ^^^^^^^^^^^^^ source.just
#                            ^^ source.just keyword.operator.assignment.just
#                              ^ source.just
#                               ^ source.just string.quoted.double.just string.quoted.double.just
#                                ^^ source.just string.quoted.double.just constant.character.escape.just
#                                  ^ source.just string.quoted.double.just
>string-with-newline         := "\n"
#^^^^^^^^^^^^^^^^^^^ source.just variable.other.just
#                   ^^^^^^^^^ source.just
#                            ^^ source.just keyword.operator.assignment.just
#                              ^ source.just
#                               ^ source.just string.quoted.double.just string.quoted.double.just
#                                ^^ source.just string.quoted.double.just constant.character.escape.just
#                                  ^ source.just string.quoted.double.just
>string-with-carriage-return := "\r"
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just variable.other.just
#                           ^ source.just
#                            ^^ source.just keyword.operator.assignment.just
#                              ^ source.just
#                               ^ source.just string.quoted.double.just string.quoted.double.just
#                                ^^ source.just string.quoted.double.just constant.character.escape.just
#                                  ^ source.just string.quoted.double.just
>string-with-double-quote    := "\""
#^^^^^^^^^^^^^^^^^^^^^^^^ source.just variable.other.just
#                        ^^^^ source.just
#                            ^^ source.just keyword.operator.assignment.just
#                              ^ source.just
#                               ^ source.just string.quoted.double.just string.quoted.double.just
#                                ^^ source.just string.quoted.double.just constant.character.escape.just
#                                  ^ source.just string.quoted.double.just
>string-with-slash           := "\\"
#^^^^^^^^^^^^^^^^^ source.just variable.other.just
#                 ^^^^^^^^^^^ source.just
#                            ^^ source.just keyword.operator.assignment.just
#                              ^ source.just
#                               ^ source.just string.quoted.double.just string.quoted.double.just
#                                ^^ source.just string.quoted.double.just constant.character.escape.just
#                                  ^ source.just string.quoted.double.just
>string-with-no-newline      := "\
#^^^^^^^^^^^^^^^^^^^^^^ source.just variable.other.just
#                      ^^^^^^ source.just
#                            ^^ source.just keyword.operator.assignment.just
#                              ^ source.just
#                               ^ source.just string.quoted.double.just string.quoted.double.just
#                                ^^ source.just string.quoted.double.just
>"
#^ source.just string.quoted.double.just
>
>escapes := """\t\n\r\"\\"""
#^^^^^^^ source.just variable.other.just
#       ^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^ source.just
#           ^^^ source.just string.quoted.double.indented.just string.quoted.double.indented.just
#              ^^ source.just string.quoted.double.indented.just constant.character.escape.just
#                ^^ source.just string.quoted.double.indented.just constant.character.escape.just
#                  ^^ source.just string.quoted.double.indented.just constant.character.escape.just
#                    ^^ source.just string.quoted.double.indented.just constant.character.escape.just
#                      ^^ source.just string.quoted.double.indented.just constant.character.escape.just
#                        ^^^ source.just string.quoted.double.indented.just
>escapes := '\t\n\r\"\\'
#^^^^^^^ source.just variable.other.just
#       ^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^ source.just
#           ^ source.just string.quoted.single.just string.quoted.single.just
#            ^^^^^^^^^^ source.just string.quoted.single.just
#                      ^ source.just string.quoted.single.just
>escapes := '''\t\n\r\"\\'''
#^^^^^^^ source.just variable.other.just
#       ^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^ source.just
#           ^^^ source.just string.quoted.single.indented.just string.quoted.single.indented.just
#              ^^^^^^^^^^ source.just string.quoted.single.indented.just
#                        ^^^ source.just string.quoted.single.indented.just
>
># Multi-line
#^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>single := '
#^^^^^^ source.just variable.other.just
#      ^ source.just
#       ^^ source.just keyword.operator.assignment.just
#         ^ source.just
#          ^ source.just string.quoted.single.just string.quoted.single.just
>hello
#^^^^^^ source.just string.quoted.single.just
>'
#^ source.just string.quoted.single.just
>
>double := "
#^^^^^^ source.just variable.other.just
#      ^ source.just
#       ^^ source.just keyword.operator.assignment.just
#         ^ source.just
#          ^ source.just string.quoted.double.just string.quoted.double.just
>goodbye
#^^^^^^^^ source.just string.quoted.double.just
>"
#^ source.just string.quoted.double.just
>
># Indented
#^^^^^^^^^^ source.just comment.line.number-sign.just
>
>x := '''
#^ source.just variable.other.just
# ^ source.just
#  ^^ source.just keyword.operator.assignment.just
#    ^ source.just
#     ^^^ source.just string.quoted.single.indented.just string.quoted.single.indented.just
>  foo
#^^^^^^ source.just string.quoted.single.indented.just
>  bar
#^^^^^^ source.just string.quoted.single.indented.just
>'''
#^^^ source.just string.quoted.single.indented.just
>
>y := """
#^ source.just variable.other.just
# ^ source.just
#  ^^ source.just keyword.operator.assignment.just
#    ^ source.just
#     ^^^ source.just string.quoted.double.indented.just string.quoted.double.indented.just
>  abc
#^^^^^^ source.just string.quoted.double.indented.just
>    wuv
#^^^^^^^^ source.just string.quoted.double.indented.just
>  xyz
#^^^^^^ source.just string.quoted.double.indented.just
>"""
#^^^ source.just string.quoted.double.indented.just
>
># Expanded
#^^^^^^^^^^ source.just comment.line.number-sign.just
>
># TODO: should there be special highlighting for these strings?
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>foobar := x"~/$FOO/${BAR}"
#^^^^^^ source.just variable.other.just
#      ^ source.just
#       ^^ source.just keyword.operator.assignment.just
#         ^ source.just
#          ^ source.just string.quoted.double.just constant.character.expanded.just
#           ^ source.just string.quoted.double.just string.quoted.double.just
#            ^^^^^^^^^^^^^ source.just string.quoted.double.just
#                         ^ source.just string.quoted.double.just
>foobar := x"""~/$FOO/${BAR}"""
#^^^^^^ source.just variable.other.just
#      ^ source.just
#       ^^ source.just keyword.operator.assignment.just
#         ^ source.just
#          ^ source.just string.quoted.double.indented.just constant.character.expanded.just
#           ^^^ source.just string.quoted.double.indented.just string.quoted.double.indented.just
#              ^^^^^^^^^^^^^ source.just string.quoted.double.indented.just
#                           ^^^ source.just string.quoted.double.indented.just
>foobar := x'~/$FOO/${BAR}'
#^^^^^^ source.just variable.other.just
#      ^ source.just
#       ^^ source.just keyword.operator.assignment.just
#         ^ source.just
#          ^ source.just string.quoted.single.just constant.character.expanded.just
#           ^ source.just string.quoted.single.just string.quoted.single.just
#            ^^^^^^^^^^^^^ source.just string.quoted.single.just
#                         ^ source.just string.quoted.single.just
>foobar := x'''~/$FOO/${BAR}'''
#^^^^^^ source.just variable.other.just
#      ^ source.just
#       ^^ source.just keyword.operator.assignment.just
#         ^ source.just
#          ^ source.just string.quoted.single.indented.just constant.character.expanded.just
#           ^^^ source.just string.quoted.single.indented.just string.quoted.single.indented.just
#              ^^^^^^^^^^^^^ source.just string.quoted.single.indented.just
#                           ^^^ source.just string.quoted.single.indented.just
>